runtime/testy/studentAPI.py

```diff
@@ -74,32 +74,45 @@ def get_value(self, name, gamepad_number=0):
     raise StudentAPIKeyError()
 
 class Robot(StudentAPI):
-  deviceType_to_validParams = {
-    10 : ["duty_cycle", "enable"], 
+  deviceType_to_writeParams = {
+    5  : ["s1", "s2", "s3", "s4"],
+    7  : ["servo0", "enable0", "servo1", "enable1"],
+    10 : ["enable", "command_state", "duty_cycle", "pid_pos_setpoint", "pid_pos_kp", "pid_pos_ki", "pid_pos_kd", "current_thresh", "enc_pos"],
+  }
+  deviceType_to_readParams = {
+    0  : ["switch0", "switch1", "switch2"],
+    1  : ["left", "center", "right"],
+    2  : ["pot0", "pot1", "pot2"],
+    10 : ["duty_cycle", "pid_pos_setpoint", "pid_pos_kp", "pid_pos_ki", "pid_pos_kd", "current_thresh", "enc_pos", "enc_vel"],
   }
   param_to_valid_values = {
-    "duty_cycle" : [float, -1, 1],
-    "enable" : [bool],
+    "servo0" : [[int], 0, 180],
```

Comment:
Your check values method only expects types to be floats or bools. Setting/getting servo values won't be checked correctly then